build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,266 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { MonoRepo, Package } from "@fluidframework/build-tools";
+import async from "async";
+import chalk from "chalk";
+import execa from "execa";
+import { ResetMode } from "simple-git";
+
+import { findPackageOrReleaseGroup, packageOrReleaseGroupArg } from "../../args.js";
+import {
+	BaseCommand,
+	type Context,
+	Repository,
+	getPreReleaseDependencies,
+} from "../../library/index.js";
+
+type CheckFunction = (
+	context: Context,
+	rgOrPkg: MonoRepo | Package,
+	print?: boolean,
+) => Promise<CheckResult>;
+
+interface CheckResult {
+	result: boolean;
+	message?: string;
+	fixCommand?: string;
+	fatal?: boolean;
+}
+
+async function checkNoLocalChanges(context: Context): Promise<CheckResult> {
+	const status = await context.gitRepo.getStatus();
+	if (status !== "") {
+		return {
+			result: false,
+			message: "There are some local changes. The branch should be clean.",
+			fixCommand: "git stash",
+			fatal: true,
+		};
+	}
+
+	return { result: true };
+}
+
+/**
+ * Returns true if all the dependencies are installed for the package/release group.
+ */
+async function checkDepsInstalled(
+	context: Context,
+	rgOrPkg: MonoRepo | Package,
+): Promise<CheckResult> {
+	const packagesToCheck = rgOrPkg instanceof MonoRepo ? rgOrPkg.packages : [rgOrPkg];
+
+	// eslint-disable-next-line @typescript-eslint/no-misused-promises -- The usage appears correct so mayeb a false positive
+	const checkDeps: async.AsyncBooleanIterator<Package> = async (
+		pkg: Package,
+	): Promise<boolean> => {
+		return pkg.checkInstall(false);
+	};
+
+	// I think the unicorn/no-array-method-this-argument lint error is a false positive. As far as I can tell, the
+	// checkDeps function is defined correctly and has the correct type. It appears to work as expected at runtime.
+	// eslint-disable-next-line unicorn/no-array-method-this-argument
+	const installed = await async.every(packagesToCheck, checkDeps);
+	if (!installed) {
+		return {
+			result: false,
+			message: "Some dependencies aren't installed. Try installing dependencies manually.",
+			fixCommand: "pnpm install",
+		};
+	}
+
+	return { result: true };
+}
+
+/**
+ * Returns true if the local git repo has a remote for the microsoft/FluidFramework repo.
```

Comment:
Same issue with above function: does not return a boolean, and does not reference CheckResult. Same applies to the ones below as well.